c++/test/TestWriter.cc

```diff
@@ -1996,5 +1996,5 @@ namespace orc {
     }
   }
 
-  INSTANTIATE_TEST_CASE_P(OrcTest, WriterTest, Values(FileVersion::v_0_11(), FileVersion::v_0_12()));
+  INSTANTIATE_TEST_CASE_P(OrcTest, WriterTest, Values(FileVersion::v_0_11(), FileVersion::v_0_12(), FileVersion::UNSTABLE_PRE_2_0()));
```

Comment:
Should we generate some ORC files and add some test cases for cross-language compatibility? We have had issues when Java ORC reader cannot understand bloom filter produced by the C++ writer (which is caught by @stiga-huang). We may be safely assured when test cases report that the Java ORC reader can correctly read UNSTABLE_PRE_2_0 file created by the C++ ORC writer and vice versa. @dongjoon-hyun @stiga-huang @coderex2522

Reply:
+1 for having that. BTW, what is the ORC issue ID for that bloom filter compatibility issue?

Reply:
> at is the ORC issue ID for that

https://issues.apache.org/jira/browse/ORC-1024

Reply:
Thank you.

Reply:
> Should we generate some ORC files and add some test cases for cross-language compatibility? We have had issues when Java ORC reader cannot understand bloom filter produced by the C++ writer (which is caught by @stiga-huang). We may be safely assured when test cases report that the Java ORC reader can correctly read UNSTABLE_PRE_2_0 file created by the C++ ORC writer and vice versa. @dongjoon-hyun @stiga-huang @coderex2522

The [PR](https://github.com/apache/orc/pull/1062) has been tested using c++ ORC reader to read UNSTABLE_PRE_2_0 file created by the java ORC writer, so here only adds the test about java reader reads the file written by c++ writer.